java-rest-service/pom.xml

```diff
@@ -46,21 +46,75 @@
             <optional>true</optional>
         </dependency>
 
+        <!-- OpenAPI -->
         <dependency>
             <groupId>org.springdoc</groupId>
             <artifactId>springdoc-openapi-starter-webmvc-api</artifactId>
             <version>${springdoc.version}</version>
         </dependency>
 
+        <!-- Database -->
+        <!-- #IF(#databaseType == 'postgres') -->
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+        <!-- #ENDIF -->
+        <!-- #IF(#databaseType == 'mysql') -->
+        <dependency>
+            <groupId>com.mysql</groupId>
+            <artifactId>mysql-connector-j</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+        <!-- #ENDIF -->
+        <!-- #IF(#databaseIntegrationTestType == 'in-memory') -->
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <!-- #ENDIF -->
+
         <!-- Infrastructure -->
+        <!-- #IF(#databaseMigrationTool == 'flyway') -->
         <dependency>
             <groupId>org.flywaydb</groupId>
             <artifactId>flyway-core</artifactId>
         </dependency>
+        <!-- #ENDIF -->
+        <!-- #IF(#databaseMigrationTool == 'flyway' && #databaseType == 'postgres') -->
```

Comment:
We _could_ remove the first condition and extend the check for flyway that starts at line 80 until the end, using nested directives

Reply:
@ericbottard  I tried to nest the `#IF` directives but couldn't get it to work - do we have this documented, I'm not exactly sure what to expect.

I tried something like this but that did not fully work
```
#IF(#tests)
This is nested test:
#IF(#javaVersion == '17')
Java 17
#ENDIF
#IF(#javaVersion == '21')
Java 21
#ENDIF
#ENDIF
```
I got the java version shown with tests set to false

This worked as I would expect:
```
#IF(#tests)
This is explicit test:
#IF(#tests && #javaVersion == '17')
Java 17
#ENDIF
#IF(#tests && #javaVersion == '21')
Java 21
#ENDIF
#ENDIF
```

Reply:
Created https://github.com/pivotal/acc-engine/issues/481